plugins/kotlin/src/main/java/org/quiltmc/quiltflower/kotlin/expr/KFunctionExprent.java

```diff
@@ -20,17 +29,31 @@ public TextBuffer toJava(int indent) {
   
     switch(getFuncType()){
       case TERNARY:
+        Exprent condition = lstOperands.get(0);
+        Exprent ifTrue = lstOperands.get(1);
+        Exprent ifFalse = lstOperands.get(2);
+
+        if (
+          condition instanceof KFunctionExprent && ((KFunctionExprent) condition).getFuncType() == FunctionType.INSTANCEOF
+          && ifTrue instanceof KFunctionExprent && ((KFunctionExprent) ifTrue).getFuncType() == FunctionType.CAST
+          && ifFalse.getExprType() == VarType.VARTYPE_NULL
+        ) {
+          // Safe cast
+          TextBuffer cast = ifTrue.toJava(indent);
+          buf.append(cast.convertToStringAndAllowDataDiscard().replace(" as ", " as? "));
```

Comment:
What's this for? convertToStringAndAllowDataDiscard is not intended for use within `toJava()` code like this- there is likely a better way to solve this